### PR TITLE
Move to flakes and move to Hydra

### PR DESCRIPTION
**Pull request by t4ccer**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,126 @@
+{
+  description = "nft-marketplace-server";
+
+  inputs = {
+    haskell-nix.url = "github:input-output-hk/haskell.nix/fa2fa131fe15e630c91ab4078d12eb32c41f934b";
```
Comment (rynoV) [inline flake.nix:5]:
```suggestion
    haskell-nix.url = "github:input-output-hk/haskell.nix/fa2fa131fe15e630c91ab4078d12eb32c41f934b";
    flake-compat = {
      url = "github:edolstra/flake-compat";
      flake = false;
    };
```

**Review by rynoV — CHANGES_REQUESTED** [review]
When I tried `nix develop` the first time I got

```
error: derivation '/nix/store/lwfskwclvzy74462idx5j0xw05bb1zpw-hoogle-with-packages.drv' requires non-existent output 'doc' from input derivation '/nix/store/2p4000j6s71nxlzcc8wbdxkdpw8k4xgn-Cabal-lib-Cabal-3.6.2.0-haddock.drv'
```

but when I ran it a second time it worked and I was able to run all the make targets.

Also: Can we include a `shell.nix` like this for `lorri`+`direnv` users:

```nix
(
  import (
    let
      lock = builtins.fromJSON (builtins.readFile ./flake.lock);
    in
      fetchTarball {
        url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
        sha256 = lock.nodes.flake-compat.locked.narHash;
      }
  ) {
    src = ./.;
  }
).shellNix.default
```

And also add `.envrc` to the `.gitignore`? I have these changes locally, lmk and I can push them up.

**Comment by rynoV** [comment]
Also `nix build .#check.x86_64-linux` works for me locally

**Comment by t4ccer** [comment]
> but when I ran it a second time it worked and I was able to run all the make targets.

Reproducibility... It worked every time for me, also it's common setup

wrt. `shell.nix` feel free to push if you have it set up

**Review by rynoV — APPROVED** [review]

**Pull request merged**